app/Services/Admin/AdminVaccinationService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using backend.Database;
+using backend.Dto.Requests.Admin;
+using backend.Dto.Responses;
+using backend.Dto.Responses.Admin.Vaccination;
+using backend.Exceptions;
+using backend.Helpers;
+using backend.Models.Vaccines;
+using backend.Models.Visits;
+using Microsoft.EntityFrameworkCore;
+
+namespace backend.Services.Admin
+{
+    public class AdminVaccinationService
+    {
+        private readonly DataContext dataContext;
+        private readonly Mailer mailer;
+
+        public AdminVaccinationService(DataContext dataContext, Mailer mailer)
+        {
+            this.dataContext = dataContext;
+            this.mailer = mailer;
+        }
+
+        public async Task<SuccessResponse> ChangeVaccinationSlot(int vaccinationId, int newVaccinationSlotId)
+        {
+            // Find vaccination visit with matching id
+            VaccinationModel vaccination = this.dataContext
+                .Vaccinations
+                .FirstOrThrow(vaccination => vaccination.Id == vaccinationId, new NotFoundException("Vaccination visit not found"));
+
+            // Get slot connected to found vaccination visit
+            VaccinationSlotModel currentSlot = this.dataContext
+                .VaccinationSlots
+                .FirstOrThrow(slot => slot.Id == vaccination.VaccinationSlotId, new NotFoundException("Current vaccination slot not found"));
+
+            // Get new slot
+            VaccinationSlotModel newSlot = this.dataContext
+                .VaccinationSlots
+                .FirstOrThrow(slot => slot.Id == newVaccinationSlotId, new NotFoundException("New vaccination slot not found"));
+
+            // Block concurrent access
+            Semaphores.slotSemaphore.WaitOne();
```

Comment:
please move that up, so the searching of the 'new slot' is blocked as well

Reply:
It will be fixed in #59 PR which contains similar problem.